crates/epaint/src/text/text_layout.rs

```diff
@@ -624,14 +649,21 @@ fn halign_and_justify_row(
     }
 
     placed_row.pos.x = point_scale.round_to_pixel(target_min_x);
-    let mut translate_x = -original_min_x - extra_x_per_glyph * glyph_range.0 as f32;
-
-    for glyph in &mut row.glyphs {
-        glyph.pos.x += translate_x;
-        glyph.pos.x = point_scale.round_to_pixel(glyph.pos.x);
-        translate_x += extra_x_per_glyph;
```

Comment:
Note: there was a secondary minor bug here:

previously we were distributing the `extra_x_per_glyph` among all glyphs, even those that we had chosen to be trimmed. We calculated `extra_x_per_glyph` based on `num_glyphs_in_range`, but were then multiplying it by total glyphs. We were probably getting a bit too much extra spacing.

Below, we only add more `extra_x_*` when we're operating on glyphs within the `visible_glyphs` range.